roles/openshift_health_checker/openshift_checks/efk_stack.py

```diff
@@ -0,0 +1,348 @@
+                          "so we cannot check if it works. Is a router running?")
+        elif status["kibana_route"].get("route_missing_host"):
+            errors.append("The route for the Kibana logging url has no host defined, "
+                          "which should never happen. Did something alter its definition?")
+        elif status["kibana_route"].get("bad_response"):
+            errors.append("Attempting to access the Kibana logging url returned an invalid response: ")
+            errors.append(status["kibana_route"].get("bad_response"))
+
+        if not status["curator"]["number_pods"]:
+            errors.append("Unable to find any curator pods. Is curator deployed on the platform?")
+        elif not status["curator"]["running"]:
+            errors.append("Curator has been deployed, but some or all pods are not yet ready.")
+
+        if len(errors) > 1:
+            return {"failed": True, "msg": "\n".join(errors)}
+
+        return {"changed": False, "msg": status}
+
+    def get_pods(self, task_vars):
+        """Get all pods and filter them in one pass"""
+        pods = json.loads(self.exec_oc("get pods -o json", [], task_vars))
+        es_pods = []
+        fluentd_pods = []
+        curator_pods = []
+
+        for pod in pods['items']:
+            if 'component' in pod['metadata']['labels']:
+                # Get ES pods
+                if pod['metadata']['labels']['component'] == 'es':
+                    es_pods.append(pod)
+                elif pod['metadata']['labels']['component'] == 'fluentd':
+                    fluentd_pods.append(pod)
+                elif pod['metadata']['labels']['component'] == 'curator':
+                    curator_pods.append(pod)
+
+        return es_pods, fluentd_pods, curator_pods
+
+    def check_elasticsearch(self, es_pods, task_vars):
+        """Various checks for elasticsearch"""
+        es_status = dict()
+        es_status['single_master'] = None
+        es_master_name = None
+        es_status['pods'] = {}
+
+        for pod in es_pods:
+            pod_dc = pod.get('metadata', {}).get('labels', {}).get('deploymentconfig', '')
+            if not pod_dc:
+                continue
+
+            pod_name = pod.get('metadata', {}).get('name', '')
+            es_status['pods'][pod_dc] = {}
+
+            if not pod_name:
+                continue
+
+            es_health_check = self.check_es_cluster_health(pod_name, task_vars)
+            es_status['pods'][pod_dc]['elasticsearch_health'] = es_health_check
+
+            es_disk_check = self.check_elasticsearch_diskspace(pod_name, task_vars)
+            es_status['pods'][pod_dc]['disk'] = es_disk_check
+
+            # Compare the master across all ES nodes to see if we have split brain
+            curl_cmd = "{} 'https://localhost:9200/_cat/master'".format(self.es_curl)
+            es_master = "exec -ti {} -- {}".format(pod_name, curl_cmd)
```

Comment:
Suggestion: perhaps we can hide all this into a method, we build the command in multiple places.